cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.model.cruisecontrol;
+
+import io.strimzi.api.kafka.model.balancing.CruiseControlBrokerCapacity;
+import io.strimzi.api.kafka.model.KafkaSpec;
+import io.strimzi.api.kafka.model.storage.EphemeralStorage;
+import io.strimzi.api.kafka.model.storage.JbodStorage;
+import io.strimzi.api.kafka.model.storage.PersistentClaimStorage;
+import io.strimzi.api.kafka.model.storage.SingleVolumeStorage;
+import io.strimzi.api.kafka.model.storage.Storage;
+
+import java.util.List;
+
+import static io.strimzi.operator.cluster.model.StorageUtils.parseMemorybyFactor;
+
+public class Capacity {
+    public static final int DEFAULT_BROKER_DISK_CAPACITY   = 100000;  // in MB
+    public static final int DEFAULT_BROKER_CPU_CAPACITY    = 100;     // as a percentage (0-100)
+    public static final int DEFAULT_BROKER_NW_IN_CAPACITY  = 10000;   // in KB/s
+    public static final int DEFAULT_BROKER_NW_OUT_CAPACITY = 10000;   // in KB/s
+
+    private Integer disk;
+    private Integer cpu;
+    private Integer nwIn;
+    private Integer nwOut;
+
+    public Capacity(KafkaSpec spec) {
+        CruiseControlBrokerCapacity bc = spec.getCruiseControl().getCapacity();
+
+        if (bc != null) {
+            this.disk = bc.getDisk() != null ? bc.getDisk() : generateDiskCapacity(spec.getKafka().getStorage());
+            this.cpu = bc.getCpu() != null ? bc.getCpu() : DEFAULT_BROKER_CPU_CAPACITY;
+            this.nwIn = bc.getNetworkIn() != null ? bc.getNetworkIn() : DEFAULT_BROKER_NW_IN_CAPACITY;
+            this.nwOut = bc.getNetworkOut() != null ? bc.getNetworkOut() : DEFAULT_BROKER_NW_OUT_CAPACITY;
+        } else {
+            this.disk = generateDiskCapacity(spec.getKafka().getStorage());
+            this.cpu = DEFAULT_BROKER_CPU_CAPACITY;
+            this.nwIn = DEFAULT_BROKER_NW_IN_CAPACITY;
+            this.nwOut = DEFAULT_BROKER_NW_OUT_CAPACITY;
+        }
```

Comment:
to avoid the outer `if` and writing duplicate code may I suggest something like this (removing the `bc != null`):

```java
this.disk = bc != null && bc.getDisk() != null ? bc.getDisk() : generateDiskCapacity(spec.getKafka().getStorage());
this.cpu = bc != null && bc.getCpu() != null ? bc.getCpu() : DEFAULT_BROKER_CPU_CAPACITY;
this.nwIn = bc != null && bc.getNetworkIn() != null ? bc.getNetworkIn() : DEFAULT_BROKER_NW_IN_CAPACITY;
this.nwOut = bc != null && bc.getNetworkOut() != null ? bc.getNetworkOut() : DEFAULT_BROKER_NW_OUT_CAPACITY;
```

and even here, renaming `nwIn` and `nwOut`.